Index: generalization_projection_technique/parsers/parser_eom.py
===================================================================
diff --git a/generalization_projection_technique/parsers/parser_eom.py b/generalization_projection_technique/parsers/parser_eom.py
--- a/generalization_projection_technique/parsers/parser_eom.py	
+++ b/generalization_projection_technique/parsers/parser_eom.py	
@@ -4,8 +4,7 @@
 from generalization_projection_technique.parsers.parser_qchem_tddft import search_word_line
 
 file = '../../\
-molecules/eomccsd_outs/\
-benzophenone_eomee_sto3g.out'
+molecules/eomccsd_outs/benzophenone_eomsf_sto3g.out'
 transition_cutoff = 0.8
 ref_state = '1/A'
 
@@ -324,35 +323,96 @@
     return eom_presentation_list, selected_excitation_energies_eom, selected_eom_soc_constants
 
 
-# orbital_sym = get_orbital_symmetries(file)
+def get_selectedstates_and_totalenergies(filee, states_option, init_states, sym_selection):
+    """
+    Depending on "states_option" it returns states:
+    0) initial states
+    1) All states in qchem output
+    :param filee:
+    :param states_option:
+    :param init_states:
+    :param sym_selection:
+    :return: select_states, total_energies
+    """
+    search = ['EOMIP transition ', 'EOMSF transition ', 'EOMEE transition ', 'EOMEA transition ']
+    nstates = 0
+    all_states = []
+    total_energies = {}
+
+    with open(filee, encoding="utf8") as f:
+        for line in f:
+            if any(a in line for a in search):
+                state = line.split()[-1]
+                all_states.append(state)
+                line = next(f)
+                total_energies.update({state: line.split()[3]})
+                nstates += 1
 
-eom_version = get_eom_type(file)
+            if 'State A' in line:
+                ground_state = line.split()[-1]
+                break
 
-scf_reference_energy = get_scf_energy(file)
+    select_states = []
+    if states_option == 0:
+        for i in init_states:
+            if 0 < i <= nstates:
+                check_state = str(i) + "/"
+                for state in all_states:
+                    if check_state in state:
+                        select_states.append(state)
+            else:
+                raise ValueError("The number of states selected selected must be among the total number of states "
+                                 "selected calculated in QChem. "
+                                 "Select a different number of states selected")
 
-states, total_energies = get_energies(file, eom_version)
+    elif states_option == 1:
+        select_states = all_states
 
-# allstates_excit_energies, allstates_momentums, allstates_spins, allstates_socs = get_interstate_properties(file)
+    elif states_option == 2:
+        for state in all_states:
+            if sym_selection in state:
+                select_states.append(state)
+        if select_states:
+            raise ValueError("There is not this symmetry. Change the symmetry selection.")
 
-# eom_socc = get_eom_socc_values(eom_input)
+    if ground_state in select_states:
+        select_states.remove(ground_state)
+    select_states.insert(0, ground_state)
+    return select_states, total_energies
 
-orbitals_in_transitions = get_maximum_amplitude_orbitals(file, eom_version, transition_cutoff)
 
-presentation_list = []
-row_list = []
-for transition in list(total_energies.keys()):
-    for trans in orbitals_in_transitions:
-        if trans[0]['transition'] == transition:
-            ener_excit = np.round(total_energies[transition][1] - total_energies[ref_state][1], 4)
-            for subtrans in trans:
-                presentation_list.append([total_energies[transition][0], ener_excit, subtrans['orbitals']])
-                row_list.append(transition)
-df = pd.DataFrame(np.array(presentation_list), index=row_list, columns=['Total energ', 'Excit. ener.', 'transition'])
+def get_selected_energies_spins(select_states, all_totalenergies, all_excitenergies, all_spins):
+    """
+    Get selected states energies and spins.
+    :param select_states:
+    :param all_totalenergies:
+    :param all_excitenergies:
+    :param all_spins:
+    :return: selected_states_energies, selected_states_spins
+    """
+    selected_states_energies = {}
+    selected_states_spins = {}
+    for i in select_states:
+        selected_states_energies.update({i: [float(all_totalenergies[i]), float(all_excitenergies[i])]})
+        selected_states_spins.update({i: all_spins[i]})
+    return selected_states_energies, selected_states_spins
 
-print("")
-print("------------------------")
-print("    eom-CC ANALYSIS ")
-print("------------------------")
-print("SCF   reference energy: ", scf_reference_energy)
-print()
-print(df.to_string())
+
+def get_selected_socs_orbitmoments(select_states, all_socs, all_momentums):
+    """
+    Get selected states orbital angular momentums.
+    :param select_states:
+    :param all_socs:
+    :param all_momentums:
+    :return: selected_socs, selected_momentums
+    """
+    ground_state = select_states[0]
+    select_states.remove(ground_state)
+
+    selected_socs = {}
+    selected_momentums = {}
+    for state_b in select_states:
+        interstate = str(ground_state) + "_" + str(state_b)
+        selected_socs.update({interstate: all_socs[interstate]})
+        selected_momentums.update({interstate: all_momentums[interstate]})
+    return selected_socs, selected_momentums
Index: generalization_projection_technique/parsers/parser_qchem_tddft.py
===================================================================
diff --git a/generalization_projection_technique/parsers/parser_qchem_tddft.py b/generalization_projection_technique/parsers/parser_qchem_tddft.py
--- a/generalization_projection_technique/parsers/parser_qchem_tddft.py	
+++ b/generalization_projection_technique/parsers/parser_qchem_tddft.py	
@@ -3,11 +3,11 @@
 
 state_selection = 1  # 0: use "state_ras" ; 1: use all states_selected
 initial_states = [1, 2, 3, 4, 5] # 0 is the ground state
-ground_state = 'T2'
+ground_state = '0'
 selected_multiplicity = 3
 
 file = '../../\
-molecules/tddft_outs/triplet_ref/anthracene_opt_tddft_tripletref_allmultip.out'  # str(sys.argv[1])
+molecules/tddft_outs/triplet_ref/benzophenone_tddft_tripletref_allmultip.out'  # str(sys.argv[1])
 
 #################################
 # FUNCTIONS AND CLASSES    ######
